### DISCUSSION TITLE
can I read from a json file and parse with this library?

### DISCUSSION THREAD

**Discussion by binaery-a** [nlohmann/json#2296] [Q&A] [answered]
Im trying to organize data in JSON and needed a library to use to parse JSON files. Is there example code that does that? Like how to read from a file and then parse and return what was inside?

**Comment by nlohmann** [comment, marked as answer]
This is all in the README.

**Reply by binaery-a** [reply]
I read the README, could you please point out where or what example code specifically? I am new to c++ so a lot of it is hard to find

**Reply by nlohmann** [reply]
Ok, I will add some FAQ sections.

Here is a complete example how to open a file and pretty-print it to stdout. It is also in the overview GIF.

```cpp
#include <fstream>
#include <iostream>
#include <nlohmann/json.hpp>

using json = nlohmann::json;

int main()
{
    std::ifstream f("myjson.json");
    json j = json::parse(f);
    std::cout << j.dump(2) << std::endl;
}
```

**Reply by binaery-a** [reply]
Thank you, I appreciate the help and the quick response

**Reply by nlohmann** [reply]
Glad if I could help. If you want to support me and my work on this project, you can become a [sponsor on GitHub](https://github.com/sponsors/nlohmann), or used [Paypal](http://paypal.me/nlohmann).

**Comment by binaery-a** [comment]
Hi again,

Im trying to parse a specific field such as "measured_resistance" : "30" for example using my own json file:

`{
    "test" : "string test",
    "time" : [0, 1],
    "measurements" : [
        {
            "channel_1" : "string 1",
            "channel_2" : "string 2",
            "measured_resistance" : 30,
            "R_min" : 20,
            "R_max" : 40,
            "passing" : true
        }
    ]
}`

And I read this area of the "over worked documentation" line 274:

```
#### Read from iterator range
You can also parse JSON from an iterator range; that is, from any container accessible by iterators whose content is stored as contiguous byte sequence, for instance a `std::vector<std::uint8_t>`:
```cpp
std::vector<std::uint8_t> v = {'t', 'r', 'u', 'e'};
json j = json::parse(v.begin(), v.end());
```
You may leave the iterators for the range [begin, end):
```cpp
std::vector<std::uint8_t> v = {'t', 'r', 'u', 'e'};
json j = json::parse(v);
```


Is this a good place to start and implement into my own code to parse a specific field? If not, can I get pointed in the right direction? Or how do you recommend I go about this?

**Reply by nlohmann** [reply]
See #2295 for examples. Also the README file has some examples.

**Reply by binaery-a** [reply]
Okay, I will take a look at that. Thank you!
